src/mito2/src/sst/index.rs

```diff
@@ -512,6 +535,17 @@ impl IndexBuildTask {
     ) -> Result<IndexBuildOutcome> {
         let version = &version_data.version;
         let mut indexer = self.indexer_builder.build(self.file_meta.file_id).await;
+
+        // Check SST file existence before building index to avoid failure of parquet reader.
+        if !self.check_sst_file_exists(version).await {
```

Comment:
I found that the check_sst_file_exists is an O(N) operation, although the files are already indexed by the file id.

The file meta actually contains the level of the file.